python/astro_metadata_translator/translators/sdss.py

```diff
@@ -0,0 +1,205 @@
+                  "boresight_rotation_coord": "sky",
+                  "dark_time": 0.0*u.s,  # Drift scan impies no dark time
+                  "instrument": "Imager",  # We only ever ingest data from the imager
+                  "relative_humidity": None,
+                  "temperature": None,
+                  "pressure": None,
+                  "detector_serial": "UNKNOWN",
+                  }
+
+    _trivial_map = {"exposure_time": ("EXPTIME", dict(unit=u.s)),
+                    "object": "OBJECT",
+                    "physical_filter": "FILTER",
+                    "exposure_id": "DATASUM",
+                    "visit_id": "DATASUM",
+                    "science_program": "OBJECT",  # This is the closest I can think of to a useful program
+                    "detector_name": "CCDLOC",  # This is a numeric incoding of the "slot", i.e. filter+camcol
+                    "telescope": ("TELESCOP", dict(default="2.5m")),
+                    }
+
+    #  Need a mapping from unique name to index.  The order is arbitrary.
+    detector_name_id_map = {"g1": 0, "z1": 1, "u1": 2, "i1": 3, "r1": 4, "g2": 5, "z2": 6, "u2": 7,
+                            "i2": 8, "r2": 9, "g3": 10, "z3": 11, "u3": 12, "i3": 13, "r3": 14,
+                            "g4": 15, "z4": 16, "u4": 17, "i4": 18, "r4": 19, "g5": 20, "z5": 21,
+                            "u5": 22, "i5": 23, "r5": 24, "g6": 25, "z6": 26, "u6": 27, "i6": 28, "r6": 29}
+
+    @classmethod
+    def can_translate(cls, header, filename=None):
+        """Indicate whether this translation class can translate the
+        supplied header.
+
+        Checks the INSTRUME and FILTER headers.
+
+        Parameters
+        ----------
+        header : `dict`-like
+            Header to convert to standardized form.
+        filename : `str`, optional
+            Name of file being translated.
+
+        Returns
+        -------
+        can : `bool`
+            `True` if the header is recognized by this class. `False`
+            otherwise.
+        """
+        if (cls.is_keyword_defined(header, "ORIGIN") and cls.is_keyword_defined(header, "CCDMODE") and
+                "SDSS" in header["ORIGIN"] and "DRIFT" in header["CCDMODE"]):
+            return True
+        return False
+
+    @cache_translation
+    def to_detector_unique_name(self):
+        # Docstring will be inherited. Property defined in properties.py
+        if self.is_keyword_defined(self._header, 'CAMCOL'):
+            return self.to_physical_filter()+str(self._header['CAMCOL'])
+        else:
+            raise ValueError('CAMCOL key is not definded')
+
+    @cache_translation
+    def to_detector_num(self):
+        # Docstring will be inherited. Property defined in properties.py
+        return self.detector_name_id_map[self.to_detector_unique_name()]
+
+    @cache_translation
+    def to_observation_id(self):
+        # Docstring will be inherited. Property defined in properties.py
```

Comment:
This implementation is so different that a docstring might be worth it. This is one of the problems with the docstrings in that the API is the same as the parent but you want to add one extra piece of information.